### PR TITLE
FIx bug with no Label Smoothing training

### PR DESCRIPTION
**Pull request by forjiuzhou**
There is a bug when training with no Label Smoothing, half precision.
change loss_weight = torch.ones(vocab_size) to loss_weight = torch.ones(vocab_size, dtype=torch.half)

### CODE REVIEW

**Comment by szmigacz** [comment]
Thanks for your contribution and for pointing out the issue. Unfortunately your proposed fix will break fp32 training (you always create loss_weight as a fp16 tensor). I fixed the original issue in #210, I'll close this PR.
